### PR TITLE
fix(medusa): Normalize discount code before querying DB

### PR DESCRIPTION
**Pull request by olivermrbl**
**What**
Normalize discount code before querying DB

Fixes CORE-567

### CODE REVIEW

**Review by srindom — APPROVED** [review]
LGTM!

**Comment by srindom** [comment]
Looks like one of the integration tests has a discount with lowercase code which cannot be found now with this change.

File: packages/medusa/src/models/discount.ts [hunk 1]
```diff
@@ -79,10 +79,10 @@ export class Discount extends SoftDeletableEntity {
 
   @BeforeInsert()
   private upperCaseCode(): void {
+    this.code = this.code.toUpperCase()
```
Comment (olivermrbl) [inline packages/medusa/src/models/discount.ts:82]: comment: @srindom In our discount factory, we provide an explicitly set an id leading to this function never being called

Comment (srindom) [reply packages/medusa/src/models/discount.ts:82]: 👍

**Pull request merged**